[tp] disable some assertion temporarily for torch.compile

wanchaol opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #116678
* __->__ #116573
* #116559
* #116426

Disable some runtime assertion first as it does not work with
torch.compile properly, I'll have a follow up fix in dynamo and reenable
this check again

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu @penguinwu @fegin @XilunWu @fduwjj @wz337 @tianyu-l @wconstab @yf225

wanchaol committed:
[tp] disable some assertion temporarily for torch.compile

Disable some runtime assertion first as it does not work with
torch.compile properly, I'll have a follow up fix in dynamo and reenable
this check again

[ghstack-poisoned]

Skylion007 commented on torch/distributed/tensor/parallel/style.py:
```diff
@@ -280,12 +280,14 @@ def _prepare_input_fn(self, inputs, device_mesh):
-        assert len(inputs) == len(self.input_layouts), \
-            "module inputs and input_layouts should have same length!"
+        if len(inputs) != len(self.input_layouts):
+            raise RuntimeError("module inputs and input_layouts should have same length!")
```
```suggestion
            raise AssertionError("module inputs and input_layouts should have same length!")
```

wanchaol replied:
updated thanks!

Skylion007 commented on torch/distributed/tensor/parallel/style.py:
```diff
@@ -354,12 +356,13 @@ def _prepare_out_fn(self, outputs, device_mesh):
-        assert len(outputs) == len(self.output_layouts), \
-            "module outputs and output_layouts should have same length!"
+        if len(outputs) != len(self.output_layouts):
+            raise RuntimeError("module outputs and output_layouts should have same length!")
```
```suggestion
            raise AssertionError("module outputs and output_layouts should have same length!")
```

wanchaol committed:
Update on "[tp] disable some assertion temporarily for torch.compile"


Disable some runtime assertion first as it does not work with
torch.compile properly, I'll have a follow up fix in dynamo and reenable
this check again

cc mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 awgu penguinwu fegin XilunWu fduwjj wz337 tianyu-l wconstab yf225

[ghstack-poisoned]

awgu reviewed (approved):
Sounds good!

(can consider filing an issue to track adding these back if you think it would help)

XilunWu reviewed (approved):
good to know that assertion causes trouble when interacting with torch.compile

wanchaol commented:
> Sounds good!
> 
> (can consider filing an issue to track adding these back if you think it would help)

sounds good! filed a issue https://github.com/pytorch/pytorch/issues/116712
